tc/tc_util.c

```diff
@@ -191,6 +191,56 @@ static const struct rate_suffix {
 	{ NULL }
 };
 
+int parse_percent_rate(char *rate, const char *str, char *dev)
+{
+	long max_rate_bits;
+	char percent[strlen(dev) - 1];
+	int ret;
+	double perc, rate_bits;
+
+	if (read_prop(dev, "speed", &max_rate_bits))
+		return -1;
+	strncpy(percent, str, strlen(str) - 1);
+	ret = sscanf(percent, "%lf%%", &perc);
```

Comment:
I don't think we need the `percent` variable at all, right? We can just pass `str` as the first parameter to `sscanf()`, which will parse it into a float followed by a percentage and store the float in `max_rate_bits`.